tests/misc.rs

```diff
@@ -1679,6 +1679,36 @@ assembly_source_files = %(wildcard src/arch/$(arch)/*.s)
    status:   EXIT_FAILURE,
 }
 
+test! {
```

Comment:
Sorry I didn't catch this before: The `test!` macro is kind of deprecated. I don't think it's very clear, and it's often inflexible, so anything custom often requires workarounds, so I try to use the `Test` struct and constructor for new tests (which the `test!` macro uses under the hood). So these should be converted to use the `Test` struct.

Reply:
I know you said for **new tests** but I took the liberty to convert the other tests that this PR touches as well.

Reply:
Nice, that's great. Maybe one day I'll get around to actually converting all the tests 😅 It strikes me that AI might actually be really good at this kind of tedious but complex refactoring.